lib/index.js

```diff
@@ -73,50 +73,68 @@ function highlight (sqlString, options) {
       matches.push({
         name: hl.name,
         start: match.index + boringLength,
-        length: (hl.trimEnd ? text.substr(0, text.length - hl.trimEnd) : text).length
+        length: (hl.trimEnd
+          ? text.substring(0, text.length - hl.trimEnd)
+          : text
+        ).length
       })
     }
   }
 
   const sortedMatches = matches.slice().sort((a, b) => a.start - b.start)
 
   // filter/exclude nested matches (matches within the last match)
-  const filteredMatches = []
+  const segments = []
   let upperBound = 0
   for (let i = 0; i < sortedMatches.length; i++) {
     if (sortedMatches[i].start >= upperBound) {
-      filteredMatches.push(sortedMatches[i])
+      if (sortedMatches[i].start > upperBound) {
+        segments.push({
+          name: DEFAULT_KEYWORD,
+          content: sqlString.substring(upperBound, sortedMatches[i].start)
+        })
+      }
+
+      segments.push({
+        name: sortedMatches[i].name,
+        content: sqlString.substring(
+          sortedMatches[i].start,
+          sortedMatches[i].start + sortedMatches[i].length
+        )
+      })
       upperBound = sortedMatches[i].start + sortedMatches[i].length
     }
   }
 
-  let highlighted = ''
-
-  for (let i = 0; i < filteredMatches.length; i++) {
-    const match = filteredMatches[i]
-    const nextMatch = filteredMatches[i + 1]
-
-    const stringMatch = sqlString.substr(match.start, match.length)
-
-    if (options.html) {
-      highlighted += `<span class="${options.classPrefix}${match.name}">`
-      highlighted += stringMatch
-      highlighted += '</span>'
-    } else {
-      highlighted += options.colors[match.name]
-      highlighted += stringMatch
-      highlighted += options.colors.clear
-    }
-    if (nextMatch) {
-      highlighted += sqlString.substr(match.start + match.length, nextMatch.start - (match.start + match.length))
-    } else if (sqlString.length > (match.start + match.length)) {
-      highlighted += sqlString.substr(match.start + match.length)
-    }
+  if (upperBound < sqlString.length - 1) {
+    segments.push({
+      name: DEFAULT_KEYWORD,
+      content: sqlString.substring(
+        upperBound,
+        upperBound + sqlString.length + 1
+      )
+    })
   }
+  return segments
+}
 
-  return highlighted
+function highlight (sqlString, options) {
+  options = Object.assign({}, DEFAULT_OPTIONS, options)
+
+  return getSegments(sqlString)
+    .map(({ name, content }) => {
+      if (name === DEFAULT_KEYWORD) {
+        return content
+      }
```

Comment:
Just an idea (might be bad). What if we added a color in the `DEFAULT_OPTIONS.colors` map with an empty value like this? That would allow for people to do something to the whitespace as well if the so wish (like adding transparent dots for clarity).

```js
colors: {
  [DEFAULT_KEYWORD]: '',
  // ...
}
```

Reply:
I did it. On the one hand i like the possibility, on the other hand it might be a breaking change for some users (html output).

Reply:
Hmm yes that's true. I'm not too keen on releasing yet another major version. If not too much work revert it and stick to the solution you came up with initially. I might put it behind an option in the future or wait with it until there is need for bigger API changes.

Reply:
I reverted it. I also think it is not worth it for an API break